### PR TITLE
meson: add workaround for Tiger

### PR DESCRIPTION
**Pull request by kencu**
and peg meson at 55.3 on Tiger  as updates are likely fragile
will update meson occasionally, as we do for libuv on Tiger

###### Type(s)

- [x] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 10.4
Xcode 2.5

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

So -- keeping Tiger alive -- this latest meson fix tries to work around the lack of rpath support in Tiger.

It largely works -- I think better than before, due to some helpful recent changes in meson upstream wherein they seem to be interested in using loader_path themselves for at least some of the in-tree executables.

Appreciate testing by anyone still interested. Perfection is not guaranteed .. (the opposite!) but improvements seem present.

### CODE REVIEW

File: devel/meson/files/patch-meson55-tiger-no-rpath-fix.diff [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+             install_name.append('.' + soversion)
+         install_name.append('.dylib')
+@@ -701,8 +701,8 @@ class AppleDynamicLinker(PosixDynamicLinkerMixin, DynamicLinker):
+         all_paths = mesonlib.OrderedSet([os.path.join(origin_placeholder, p) for p in processed_rpaths])
+         if build_rpath != '':
+             all_paths.add(build_rpath)
+-        for rp in all_paths:
+-            args.extend(self._apply_prefix('-rpath,' + rp))
++        #KEN for rp in all_paths:
++        #KEN    args.extend(self._apply_prefix('-rpath,' + rp))
```
Comment (kencu) [inline devel/meson/files/patch-meson55-tiger-no-rpath-fix.diff:34]: this line might possibly be happier if it was this line:
```
args.extend(self._apply_prefix('-L,' + rp))
```

**Comment by kencu** [comment]
ah, what I'd give for a nice way to debug meson...

**Comment by kencu** [comment]
The remaining issue I'm finding is when building documentation -- that still errors, so I've had to disable the documentation building on a couple of gnome ports, eg <https://github.com/kencu/TigerPorts/blob/68c148b0a916490825690fe725135f8b130de128/devel/atk/Portfile#L39>

**Review by SoapZA — APPROVED** [review]

**Comment by kencu** [comment]
Well, I would dearly love to fix the gnome scanner part in meson, but a few hours of poking around did not come up with the magic incantation so far. What does work is to set DYLD_LIBRARY_PATH during destroot to the folder containing the library, so that's what I've been doing lately.

I have one tiny bit to add -- to fix machine detection for "power macintosh" systems, and then I'll push this through.

**Pull request merged**